server/src/main/java/org/opensearch/extensions/AddSettingsUpdateConsumerRequestHandler.java

```diff
@@ -0,0 +1,91 @@
+import org.opensearch.transport.TransportService;
+
+/**
+ * Handles requests to add setting update consumers
+ *
+ * @opensearch.internal
+ */
+public class AddSettingsUpdateConsumerRequestHandler {
+
+    private static final Logger logger = LogManager.getLogger(AddSettingsUpdateConsumerRequestHandler.class);
+
+    private final ClusterService clusterService;
+    private final TransportService transportService;
+    private final String updateSettingsRequestType;
+
+    /**
+     * Instantiates a new Add Settings Update Consumer Request Handler with the {@link ClusterService} and {@link TransportService}
+     *
+     * @param clusterService the cluster service used to extract cluster settings
+     * @param transportService the node's transport service
+     * @param updateSettingsRequestType the update settings request type
+     */
+    public AddSettingsUpdateConsumerRequestHandler(
+        ClusterService clusterService,
+        TransportService transportService,
+        String updateSettingsRequestType
+    ) {
+        this.clusterService = clusterService;
+        this.transportService = transportService;
+        this.updateSettingsRequestType = updateSettingsRequestType;
+    }
+
+    /**
+     * Handles a {@link AddSettingsUpdateConsumerRequest}.
+     *
+     * @param addSettingsUpdateConsumerRequest  The request to handle.
+     * @return  A {@link ExtensionBooleanResponse} indicating success.
+     * @throws Exception if the request is not handled properly.
+     */
+    TransportResponse handleAddSettingsUpdateConsumerRequest(AddSettingsUpdateConsumerRequest addSettingsUpdateConsumerRequest)
+        throws Exception {
+
+        boolean status = true;
+        List<WriteableSetting> extensionComponentSettings = addSettingsUpdateConsumerRequest.getComponentSettings();
+        DiscoveryExtension extensionNode = addSettingsUpdateConsumerRequest.getExtensionNode();
+
+        try {
+            for (WriteableSetting extensionComponentSetting : extensionComponentSettings) {
+
+                // Extract setting and type from writeable setting
+                Setting<?> setting = extensionComponentSetting.getSetting();
+                WriteableSetting.SettingType settingType = extensionComponentSetting.getType();
+
+                // Register setting update consumer with callback method to extension
+                clusterService.getClusterSettings().addSettingsUpdateConsumer(setting, (data) -> {
+                    logger.info("Sending extension request type: " + updateSettingsRequestType);
+                    UpdateSettingsResponseHandler updateSettingsResponseHandler = new UpdateSettingsResponseHandler();
+                    transportService.sendRequest(
+                        extensionNode,
+                        updateSettingsRequestType,
+                        new UpdateSettingsRequest(settingType, setting, data),
+                        updateSettingsResponseHandler
+                    );
```

Comment:
We are sending request to just one node here but it should be a broadcast request to all the nodes who registered the setting. No need to handle it in this PR but let's create a follow up issue.

Reply:
@dbwiddis As per your comment [here](https://github.com/opensearch-project/OpenSearch/pull/4519#discussion_r973215209), the future plan for registering settings within SettingsModule involves modifying the Setting key with the extension node's unique ID, so any settings that it would like to register is uniquely mapped to the origin extension node. If we plan to broadcast setting updates to each extension that registered  the same setting, how should we handle registering settings in such a way that these Setting keys don't conflict?

Reply:
https://github.com/opensearch-project/opensearch-sdk-java/issues/157